modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/documentation/SecureXMLHelper.java

```diff
@@ -0,0 +1,220 @@
+                        .orElse(defaultValue);
+    }
+
+    public static String secureStAX(String xml, List<String> keywords, List<String> whitelist)
+            throws XMLStreamException {
+        if (!isChangeNeeded(xml, keywords)) {
+            return xml;
+        }
+        try (InputStream stream = new ByteArrayInputStream(xml.getBytes());
+                OutputStream output = new ByteArrayOutputStream()) {
+
+            XMLEventReader reader = inputFactory.createXMLEventReader(stream);
+            XMLEventWriter writer = outputFactory.createXMLEventWriter(output);
+
+            boolean skipContent = false;
+            while (reader.hasNext()) {
+                XMLEvent event = reader.nextEvent();
+                if (skipContent && event.isCharacters()) {
+                    writer.add(eventFactory.createCharacters(SECRET_VALUE));
+                    skipContent = false;
+                    continue;
+                }
+                if (event.isStartElement()) {
+                    StartElement el = event.asStartElement();
+                    String name = el.getName().getLocalPart();
+                    if (matches(name, keywords, whitelist)) {
+                        skipContent = true;
+                    }
+                    writer.add(eventFactory.createStartElement(el.getName(), null, null));
+                    Iterator<Attribute> attrIt = el.getAttributes();
+                    while (attrIt.hasNext()) {
+                        Attribute attr = attrIt.next();
+                        String attrName = attr.getName().getLocalPart();
+                        String value = attr.getValue();
+                        if (matches(attrName, keywords, whitelist)) {
+                            // replace attribute value
+                            writer.add(eventFactory.createAttribute(attrName, SECRET_VALUE));
+                        } else {
+                            writer.add(eventFactory.createAttribute(attrName, value));
+                            if (matches(value, keywords, whitelist)) {
+                                // replace node content
+                                skipContent = true;
+                            }
+                        }
+                    }
+                } else if (event.isStartDocument()) {
+                    if (((StartDocument) event).getVersion() == null) {
+                        // skip it, it's been added by the reader processing
+                        continue;
+                    } else {
+                        writer.add(event);
+                        writer.add(eventFactory.createCharacters("\n"));
+                    }
+                } else {
+                    writer.add(event);
+                }
+
+                if (event.isEndElement()) {
+                    skipContent = false;
+                }
+            }
+            writer.flush();
+            writer.close();
+
+            String all = output.toString();
```

Comment:
Define `ByteArrayOutputStream output` above, and use `output.toString(UTF_8)` to convert to string. Charset is important.